### PR TITLE
fix(parser): allow whitespace between an entity/typed-value type name and its '('

### LINKED ISSUE #3789: Typed-value and entity regexes still require the type name and '(' to be adjacent (TS siblings of #3205)
#3205 fixed the Rust tokenizer so a typed value wrapped across whitespace between its type name and '(' parses. The same adjacency assumption survives in the TypeScript half and in the export rewriter, so the two engines now disagree on the same file:

- packages/parser/src/entity-extractor.ts ~76: the entity regex allows whitespace around '=' but not before '(' (`#5=IFCSURFACESTYLERENDERING\r\n(#4,0.);` returns null from extractEntity, so properties/quantities/materials/units/georef never see the entity).
- packages/parser/src/entity-extractor.ts ~215: the typed-value regex cannot cross whitespace (`IFCPOSITIVELENGTHMEASURE\r\n(1.)` falls through to a plain string); downstream packages/parser/src/unit-extractor.ts ~274 then replaces an unreadable ValueComponent with conversionValue 1.0, which scales geometry wrongly for an inch-based IFCCONVERSIONBASEDUNIT.
- packages/export/src/unit-normalize.ts ~230 TYPED_MEASURE_RE skips the wrapped literal, leaving an internally inconsistent normalized file.
- apps/viewer/src/components/viewer/properties/raw-step-format.ts ~148: isInlineEditableToken misses the wrapped form, so a no-op edit writes back a quoted literal with an embedded newline.

Rust side follow-up from the same review: an EMPTY argument list separated by whitespace or a comment still fails in parse_entity, list_at_depth and the typed value (`#1=IFCX(\r\n);`, `#1=IFCTABLE('a',( ));`, `#1=IFCX(IFCLABEL(\r\n));`, `(/* empty */)`): one helper `delimited(pair(char('('), ws), separated_list0(..), pair(ws, char(')')))` at the three sites closes every case and was verified to leave passing ASTs byte-identical.

### PR DESCRIPTION
**Pull request by BIMvoice**
## Summary

Closes #3789

#3205 fixed the Rust tokenizer so a typed value wrapped across whitespace between its type name and `(` still parses. The same adjacency assumption survived in the TS siblings and, for the empty-argument-list shape, in three Rust tokenizer call sites.

**TS (packages/parser, packages/export, apps/viewer):**
- `EntityExtractor.extractEntity`'s entity regex allowed whitespace around `=` but not before the args `(`, so a wrapped record (`#5=IFCSURFACESTYLERENDERING\r\n(#4,0.);`) returned `null` — the entity was invisible to every extractor keyed on it (properties, quantities, materials, units, georef).
- `EntityExtractor.parseAttributeValue`'s typed-value regex had the same gap, so a wrapped typed value (`IFCPOSITIVELENGTHMEASURE\r\n(1.)`) fell through to a plain string, which then made `unit-extractor.ts` default an unreadable `ValueComponent` to `conversionValue 1.0` instead of the real one.
- `export/unit-normalize.ts`'s `TYPED_MEASURE_RE` skipped a wrapped literal during unit rewriting, leaving the normalized file internally inconsistent.
- `viewer/raw-step-format.ts`'s `isInlineEditableToken` misclassified a wrapped typed value as editable, so a no-op edit wrote back a quoted literal with an embedded newline.

**Rust (rust/core/src/parser/tokenizer.rs):** `parse_entity`, `list_at_depth` and `typed_value_at_depth` all failed on an EMPTY argument list separated from its `)` by whitespace or a comment (`#1=IFCX(\r\n);`, `#1=IFCTABLE('a',( ));`, `(/* empty */)`) — `separated_list0` consumes nothing on zero items, so a bare `char(')')` never advanced past the trailing trivia. Fixed by putting `ws` on both sides of the parens at all three sites.

Every copy of the shape was enumerated by grepping both TS and Rust for the entity/typed-value regex pattern; the only production sites are the four listed above plus the three Rust call sites. A handful of test-file-local regexes (`packages/parser/test/*.test.ts`, `apps/viewer/.../treeDataBuilder.test.ts`) reuse a tighter `#(\d+)\s*=\s*(\w+)\(` shape for local test scaffolding only — not production code, left untouched.

## Test plan

- [x] RED: each fix has a failing-first regression test quoting the exact wrapped-form example from the issue.
- [x] Two-way: each fix also has an explicit rejection test (non-whitespace junk between type name and `(` must NOT be treated as adjacency) that stays green.
- [x] Mutation: widened each fixed pattern back toward the old permissive shape (`\s*` → `.*`, or a permissive `take_until`) and confirmed the rejection tests fail under the mutation, then reverted.
- [x] `cargo test -p ifc-lite-core` — 235 + 5 + ... all green (including 5 new tokenizer tests).
- [x] `cargo clippy -p ifc-lite-core --all-targets -- -D warnings` — clean.
- [x] `pnpm test --filter @ifc-lite/parser --filter @ifc-lite/export` — 1096+ tests green (2 new suites).
- [x] Targeted viewer run: `apps/viewer/src/components/viewer/properties/raw-step-format.test.ts` — 48/48 green.
- [x] `pnpm typecheck` (parser, export, viewer, viewer-core) — clean.
- [x] `node scripts/check-module-size.mjs` / `node scripts/check-test-wiring.mjs` — OK.
- [x] Changeset added (`@ifc-lite/parser` + `@ifc-lite/export`, patch).

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved STEP parsing when entity or typed-value names are separated from opening parentheses by line breaks, whitespace, or block comments.
  * Correctly handles empty argument lists with trailing whitespace or comments.
  * Fixed export, relationship, context, and unit-processing workflows that could previously misread wrapped records or apply incorrect scaling.
  * Preserved correct handling of quoted strings, malformed comments, and unsupported separators.
* **Performance**
  * Improved parsing reliability for long whitespace and comment sequences.

### CODE REVIEW

File: rust/core/src/parser/tokenizer_tests.rs [hunk 1]
```diff
@@ -282,3 +282,86 @@ fn test_parse_entity_typed_value_wrapped_with_crlf() {
...
+/// swallow the invalid byte as if it were trivia.
+#[test]
+fn test_list_rejects_junk_where_only_whitespace_is_allowed() {
+    let result = list(b"(x)");
+    // `x` is not `ws`/comment and not a valid token start handled by this
+    // grammar as a bare identifier, so this must NOT parse as an empty list.
+    match result {
+        Ok((_, Token::List(items))) => {
+            assert!(!items.is_empty(), "must not silently parse as an empty list");
+        }
```
[the remark this answers is not in the document]
Comment (louistrue) [reply rust/core/src/parser/tokenizer_tests.rs]: Fixed, and checked in the right order. I first confirmed the widened tokenizer really does reject it rather than assuming: `cargo test -p ifc-lite-core --lib test_list_rejects_junk` passes with `assert!(list(b"(x)").is_err())`, so the either/or match was hiding a verdict the parser already gives. The test now asserts that outright. Landed in 8e8104f, together with a shared Rust/TS trivia fixture that pins the same rejection alongside U+00A0, `//` and an unterminated comment.

File: packages/export/src/reference-collector.ts [hunk 2]
```diff
@@ -39,6 +39,14 @@ import { splitTopLevelArgs } from './step-argument-parser.js';
 import { INFRASTRUCTURE_TYPES, PRODUCT_TYPES, collectDescendantNames } from './entity-type-sets.js';
 export { INFRASTRUCTURE_TYPES, PRODUCT_TYPES, collectDescendantNames };
 
+/**
+ * `#N=TYPE(...)` record split into prefix/args/suffix, with STEP trivia
+ * (whitespace and/or a `/* ... *​/` comment, #3789) tolerated between the
+ * type name and `(` — same adjacency fix as `entity-extractor.ts`'s
+ * `extractEntity`. Shared by both line rewriters below.
+ */
+const RECORD_PREFIX_RE = new RegExp(`^(#\\d+\\s*=\\s*\\w+${STEP_TRIVIA}\\()([\\s\\S]*)(\\)\\s*;)\\s*$`);
```
[the remark this answers is not in the document]
Comment (louistrue) [reply packages/export/src/reference-collector.ts]: Confirmed and fixed in 692c717ba. `entityType` was sliced back out of the prefix capture, so a commented record produced `IFCRELCONNECTSSTRUCTURALMEMBER/* c */`, which matches no `isOptionalTrailingRef` entry and sent the optional 10th attribute down the general withhold path, dropping the whole line rather than rewriting the ref to `$`. The type name now has its own capture (1 prefix, 2 type, 3 args, 4 suffix) and `extractRelationshipRefGroupsIndexed` reads the shifted args index. Proven by running the four new tests against the pre-fix file: the comment case failed with `expected null to be '#1=IFCRELCONNECTSSTRUCTURALMEMBER/* c */(...,$);'` while the wrapped and adjacent controls were already green.

File: packages/export/src/merged-subcontext.test.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    expect(groups.size).toBe(2);
+    for (const ids of groups.values()) {
+      expect(ids.length).toBe(1);
+    }
+  });
+
+  it('two-way rule: non-whitespace, non-comment junk before "(" still reads as unreadable (both fall in the same empty-key bucket)', () => {
+    const store = buildStore([
+      [10, 'IFCGEOMETRICREPRESENTATIONSUBCONTEXTX', `#10=IFCGEOMETRICREPRESENTATIONSUBCONTEXTX(${BODY});`],
+      [11, 'IFCGEOMETRICREPRESENTATIONSUBCONTEXTX', `#11=IFCGEOMETRICREPRESENTATIONSUBCONTEXTX(${AXIS});`],
```
[the remark this answers is not in the document]
Comment (louistrue) [reply packages/export/src/merged-subcontext.test.ts]: Correct: `\w+` absorbed the trailing `X` into the type name, so both records stayed readable with distinct keys and the test passed for the opposite reason to the one its name claims. Switched to `IFCGEOMETRICREPRESENTATIONSUBCONTEXT#5(`, where `#` is neither a word character nor STEP trivia, so `RECORD_RE` genuinely fails and `getStepAttr` returns null for both, and flipped the assertion to one bucket containing `[10, 11]`.
